src/os/bluestore/BlueFS.cc

```diff
@@ -1239,17 +1270,23 @@ void BlueFS::_fsync(FileWriter *h)
 {
   dout(10) << __func__ << " " << h << " " << h->file->fnode << dendl;
   _flush(h, true);
-  _flush_wait(h);
-  if (h->file->dirty) {
-    dout(20) << __func__ << " file metadata is dirty, flushing log on "
-	     << h->file->fnode << dendl;
-    _flush_log();
-    assert(!h->file->dirty);
+  uint64_t old_dirty_seq = h->file->dirty_seq;
+  lock.unlock();
+  wait_for_aio(h);
```

Comment:
Right now we have the following test cases,
Test case 1:
1. One writer for single(keeps appending to the file).
2. Couple of threads, to generate more files(new).
Both threads keep doing fsync from both. 

Test case 2:
Have multiple threads to generate new files. will keep doing fsync.


Will add the following

Test case 3: 
Have couple of threads to do append only or create new files and have a sync_metadata thread.
This will generate log compaction once the threshould is hit.

If you have something in mind i can add them. This covers most of the uses cases coming to bluefs.